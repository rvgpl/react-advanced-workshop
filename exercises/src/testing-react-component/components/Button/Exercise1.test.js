import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Button from ".";

// Exercise #1
describe("Rendering", () => {
  // A sample test.
  it("Should render a button", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper.find('button').length).toEqual(1);
  });

  it("Should render defaults", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("Render based on size", () => {
    it("Small Button", () => {
      const wrapper = shallow(<Button size="small"/>);
      expect(wrapper.find('.btn-sm').length).toEqual(1);
    });

    it("Medium Button", () => {
      const wrapper = shallow(<Button size="medium"/>);
      expect(wrapper.find('.btn-md').length).toEqual(1);
    });

    it("Large Button", () => {
      const wrapper = shallow(<Button size="large"/>);
      expect(wrapper.find('.btn-lg').length).toEqual(1);
    });
  });

  describe("Button is disabled based on props", () => {
    it("Button should have disabled attribute", () => {
      const wrapper = shallow(<Button disabled={true}/>);
      expect(wrapper.find('button').prop('disabled')).toBeTruthy();
    });

    it("Button does not have disabled attribute", () => {
      const wrapper = shallow(<Button disabled={false}/>);
      expect(wrapper.find('button').prop('disabled')).toBeFalsy();
    });

    it("Button should have disabled class", () => {
      const wrapper = shallow(<Button disabled={true}/>);
      expect(wrapper.find('.disabled').length).toEqual(1);
    });
  });
});
