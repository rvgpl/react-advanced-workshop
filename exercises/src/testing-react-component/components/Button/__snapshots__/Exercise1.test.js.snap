// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Should render defaults 1`] = `
<button
  className="btn btn-md"
  disabled={false}
  onClick={[Function]}
>
  Button
</button>
`;
